containers/osbuild-composer/entrypoint.py

```diff
@@ -0,0 +1,136 @@
+"""entrypoint - Containerized OSBuild Composer
+
+This provides the entrypoint for a containerized osbuild-composer image. It
+spawns `osbuild-composer` on start and manages it until it exits. The main
+purpose of this entrypoint is to prepare everything to be usable from within
+a container.
+"""
+
+import os
+import socket
+import subprocess
+import sys
+
+
+#
+# Prepare Environment
+#
+# Set the environment up and prepare it for execution os osbuild-composer. This
+# does modify the calling environment!
+#
+def prepare_environment():
+    os.environ["CACHE_DIRECTORY"] = "/var/cache/osbuild-composer"
+    os.environ["STATE_DIRECTORY"] = "/var/cache/osbuild-composer"
+    os.chdir("/usr/libexec/osbuild-composer")
+
+
+#
+# Prepare Sockets
+#
+# Prepare all the API sockets that osbuild-composer expectes, and make sure to
+# pass them according to the systemd socket-activation API.
+#
+# Note that we rely on this being called early, so we get the correct FD
+# numbers assigned. We need FD-#3 onwards for compatibility with socket
+# activation (because python `subprocess.Popen` does not support renumbering
+# the sockets we pass down).
+#
+def prepare_sockets():
+    index = 3
+    sockets = []
+    fds = []
+    names = []
+
+    # osbuild-composer.socket:0
+    sock = socket.socket(socket.AF_UNIX, socket.SOCK_STREAM)
+    sock.bind("/run/weldr/api.socket")
+    sock.listen(256)
+    sockets.append(sock)
+    fds.append(sock.fileno())
+    names.append("osbuild-composer.socket")
+
+    assert(sock.fileno() == index)
+    index += 1
+
+    # osbuild-composer.socket:1
+    sock = socket.socket(socket.AF_UNIX, socket.SOCK_STREAM)
+    sock.bind("/run/osbuild-composer/job.socket")
+    sock.listen(256)
+    sockets.append(sock)
+    fds.append(sock.fileno())
+    names.append("osbuild-composer.socket")
```

Comment:
I don't think exposing the weldr API makes sense in container deployments. We could make it configurable, defaulting to off. I wouldn't mind removing it altogether, either.